src/core/nodeOps.ts

```diff
@@ -85,10 +85,24 @@ export const nodeOps: RendererOptions<TresObject, TresObject> = {
 
     if (!parent) parent = fallback as TresObject
 
+    if (child?.isObject3D) {
+      if (child?.isCamera)
+        scene?.userData?.[OBJECT_3D_USER_DATA_KEYS.REGISTER_CAMERA]?.(child)
+
+
+      if (
+        child?.onClick ||
+        child?.onPointerMove ||
+        child?.onPointerEnter ||
+        child?.onPointerLeave
+      )
+        scene?.userData?.[OBJECT_3D_USER_DATA_KEYS.REGISTER_AT_POINTER_EVENT_HANDLER]?.(child)
```

Comment:
I agree with what you say. I'm also not a big fan of the userData workaround, but I don't see any other way to access the callbacks we have in the context composable.

A custom type that fits our needs seems like a good solution for type safety and traceability for newcomers. Good idea! But be aware, that keys in userData are not in uppercase themselves. I added `keys.ts`to have all the keys at a central point and to prevent typos and collisions.

I'll tinker with the type (and probably add it) next week.